src/main/java/io/vertx/core/file/impl/FileResolver.java

```diff
@@ -236,10 +242,15 @@ private synchronized File unpackFromFileURL(URL url, String fileName, ClassLoade
     } else {
       cacheFile.mkdirs();
       String[] listing = resource.list();
-      for (String file: listing) {
-        String subResource = fileName + "/" + file;
-        URL url2 = getValidClassLoaderResource(cl, subResource);
-        unpackFromFileURL(url2, subResource, cl);
+      if (listing != null) {
+        for (String file: listing) {
+          String subResource = fileName + "/" + file;
+          URL url2 = getValidClassLoaderResource(cl, subResource);
+          if (url2 == null) {
+            throw new VertxException("Invalid resource: " + subResource);
```

Comment:
shouldn't we rather log this instead of throwing ?

Reply:
Currently we throw npe, without a clear message on which resource caused it. I kept the behavior but you also see which resource triggered it.